nup/update.go

```diff
@@ -126,6 +142,20 @@ func (u Update[T]) Diff(value T) Update[T] {
 	return u
 }
 
+// DiffPtr returns the update itself if ApplyPtr(value) does not contain a value
+// equal to the given value; otherwise it returns a no-op update. DiffPtr can be
+// used to omit extraneous updates when applying them would have no effect.
+func (u Update[T]) DiffPtr(value *T) Update[T] {
+	applied := u.ApplyPtr(value)
+	if applied == nil || value == nil {
+		if applied == value {
+			return Noop[T]()
+		}
+		return u
+	}
+	return u.Diff(*value)
```

Comment:
Would it be weird to write this as a switch? after reading it a bunch I feel like I get this, but I was definitely thrown off for a moment.

```suggestion
	switch {
	case applied == nil && value == nil:
		return Noop[T]()
	case applied == nil || value == nil:
		return u
	default:
		return u.Diff(*value)
	}

```

Reply:
That's a lot easier to grok 👍